File: Code Examples/Security_OTP_Example/src/main/java/io/hashimati/security/repository/RefreshTokenRepository.java
package io.hashimati.security.repository;


import io.hashimati.security.domains.RefreshToken;
import io.micronaut.core.annotation.NonNull;
import io.micronaut.data.jdbc.annotation.JdbcRepository;
import io.micronaut.data.model.query.builder.sql.Dialect;
import io.micronaut.data.repository.CrudRepository;

import javax.validation.constraints.NotBlank;
import java.util.Optional;

@JdbcRepository(dialect = Dialect.H2)
public interface RefreshTokenRepository extends CrudRepository<RefreshToken, Long> {
    Optional<RefreshToken> findByRefreshToken(@NonNull @NotBlank String refreshToken);

    long updateByUsername(@NonNull @NotBlank String username,
                          boolean revoked);
    void deleteByUsername(@NonNull @NotBlank String username);

}

File: Code Examples/JWT/Kotlin/JWTKotlinJDBC/src/main/kotlin/io/hashimati/security/repository/RefreshTokenRepository.kt
package io.hashimati.security.repository

import io.hashimati.security.domains.RefreshToken
import io.micronaut.core.annotation.NonNull
import io.micronaut.data.jdbc.annotation.JdbcRepository
import io.micronaut.data.model.query.builder.sql.Dialect
import io.micronaut.data.repository.CrudRepository
import java.util.*
import javax.validation.constraints.NotBlank

@JdbcRepository(dialect = Dialect.H2)
interface RefreshTokenRepository : CrudRepository<RefreshToken, Long> {
    fun findByRefreshToken(@NonNull refreshToken: @NotBlank String?): Optional<RefreshToken?>?
    fun updateByUsername(@NonNull username: @NotBlank String?, revoked: Boolean);

    fun deleteByUsername(@NonNull username: @NotBlank String?)
}